src/Neo.VM/Types/Buffer.cs

```diff
@@ -112,5 +113,14 @@ public override string ToString()
         {
             return GetSpan().TryGetString(out var str) ? $"(\"{str}\")" : $"(\"Base64: {Convert.ToBase64String(GetSpan())}\")";
         }
+
+        /// <summary>
+        /// Invalidate
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Invalidate()
```

Comment:
It's easy to miss something this way.

Reply:
If Hashcode is wrong computed, it only affect that it will call equal more frequently

Reply:
Not only result in more equal calls, but it can also lead to some bugs. For example:
Find a key in `Dictionary`, i.e. `HashTable`,  the first step is to find the keys with the same `HashCode`, if there is a cached `hashCode` and not reset after modified, this key may not be found.

Reply:
Do you think that left any place?

Reply:
> Do you think that left any place?

It's easy to forget to call `InvalidateHashCode` on later modifications, and it easy to make mistakes, similar to the current `ReferenceCounter`.